File: core/src/main/java/com/vzome/core/edits/InvertSelection.java

//(c) Copyright 2005, Scott Vorthmann.  All rights reserved.

package com.vzome.core.edits;


import com.vzome.core.editor.ChangeSelection;
import com.vzome.core.editor.Selection;
import com.vzome.core.model.Manifestation;
import com.vzome.core.model.RealizedModel;

public class InvertSelection extends ChangeSelection
{
    protected final RealizedModel mManifestations;

    public InvertSelection( Selection selection, RealizedModel model )
    {
        super( selection );
        mManifestations = model;
    }
    
    @Override
    public void perform()
    {
        for (Manifestation m : mManifestations) {
            if ( m .getRenderedObject() != null )
            {
                if ( mSelection .manifestationSelected( m ) )
                    unselect( m );
                else
                    select( m );
            }
        }
        redo();
    }

    @Override
    protected String getXmlElementName()
    {
        return "InvertSelection";
    }

}

File: online/src/worker/legacy/from-java/com/vzome/core/edits/InvertSelection.ts
import { ChangeSelection } from "../editor/api/ChangeSelection.js";
import { EditorModel } from "../editor/api/EditorModel.js";
import { RealizedModel } from "../model/RealizedModel.js";

export class InvertSelection extends ChangeSelection {
    mManifestations: RealizedModel;

    public constructor(editor: EditorModel) {
        super(editor.getSelection());
        if (this.mManifestations === undefined) { this.mManifestations = null; }
        this.mManifestations = editor.getRealizedModel();
    }

    /**
     * 
     */
    public perform() {
        for(let index=this.mManifestations.iterator();index.hasNext();) {
            let m = index.next();
            {
                if (m.isRendered()){
                    if (this.mSelection.manifestationSelected(m))this.unselect$com_vzome_core_model_Manifestation(m); else this.select$com_vzome_core_model_Manifestation(m);
                }
            }
        }
        this.redo();
    }

    /**
     * 
     * @return {string}
     */
    getXmlElementName(): string {
        return "InvertSelection";
    }
}
InvertSelection["__class"] = "com.vzome.core.edits.InvertSelection";
